pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -125,53 +121,50 @@ def _process_stream(msg):
                 self._mosek.streamtype.log, _process_stream)
 
         if self._keepfiles:
-            print("Solver log file : " + self._log_file)
+            logger.info("Solver log file: {}".format(self._log_file))
 
         for key, option in self.options.items():
             try:
-                param = ".".join(("self._mosek", key))
+                param = key.split('.')
+                if 'mosek' in param:
+                    param.remove('mosek')
+                param = getattr(self._mosek, param[0])(param[1])
                 if 'sparam' in key.split('.'):
-                    self._solver_model.putstrparam(eval(param), option)
+                    self._solver_model.putstrparam(param, option)
                 elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(eval(param), option)
+                    self._solver_model.putdouparam(param, option)
                 elif 'iparam' in key.split('.'):
                     if isinstance(option, str):
-                        if 'mosek' in option.split('.'):
-                            self._solver_model.putintparam(
-                                eval(param), eval(option))
-                        else:
-                            option = ".".join(("mosek", option))
-                            self._solver_model.putintparam(
-                                eval(param), eval(option))
+                        option = option.split('.')
+                        if 'mosek' in option:
+                            option.remove('mosek')
+                        option = getattr(self._mosek, option[0])(option[1])
                     else:
-                        self._solver_model.putintparam(eval(param), option)
+                        self._solver_model.putintparam(param, option)
```

Comment:
In case of an invalid parameter name, an `AttributeError` will be raised when I use the `getattr` method to set the value for `param`. This exception will be caught in the except statement just below, and then re-raised. So this scenario should be covered.